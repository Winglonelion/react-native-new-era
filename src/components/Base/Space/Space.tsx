import React, { memo } from 'react';
import { View } from 'react-native';

interface PropTypes {
  height?: number;
  width?: number;
}

const Space: React.FC<PropTypes> = memo(({ height = 10, width = 0 }) => {
  return <View style={{ height, width }} />;
});

export default Space;
