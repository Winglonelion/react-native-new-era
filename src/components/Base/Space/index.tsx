export { default as default } from './Space';
