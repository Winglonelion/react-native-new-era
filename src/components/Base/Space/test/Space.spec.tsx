import React from 'react';
import { shallow } from 'enzyme';
import Space, { PropTypes } from '../Space';

const setup = (propOverrides: PropTypes = {} as PropTypes) => {
  const props = { ...propOverrides };
  const wrapper = shallow(<Space {...props} />);
  return {
    props,
    wrapper,
  };
};

describe('Space check snapshot', () => {
  test('should match snapshot', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });

  test('should match snapshot with props', () => {
    const { wrapper } = setup({ height: 10, width: 10 });
    expect(wrapper).toMatchSnapshot();
  });
});
describe('Space check render correctly', () => {
  test('should contain 1 View', () => {
    const { wrapper } = setup();
    expect(wrapper.find('View')).toHaveLength(1);
  });
});
