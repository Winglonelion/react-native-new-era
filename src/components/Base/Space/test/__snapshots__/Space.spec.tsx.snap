// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Space check snapshot should match snapshot 1`] = `
<View
  style={
    Object {
      "height": 10,
      "width": 0,
    }
  }
/>
`;

exports[`Space check snapshot should match snapshot with props 1`] = `
<View
  style={
    Object {
      "height": 10,
      "width": 10,
    }
  }
/>
`;
